### PR TITLE
ElasticSearchDatabase implementation

### PR DESCRIPTION
**Pull request by fabriziofortino**
ElasticSeach (v1.4.4) can now be used as Database implementation. The configuration allows to create a new ES node (with DISK or MEMORY indexes) or to connect to an existing ES cluster through transport client connection. The latter mode is extremely useful when we need to deduplicate / link records from an existing ES instance.

See #132

### CODE REVIEW

**Comment by fabriziofortino** [comment]
Travis fails with JDK 6. Elasticsearch requires Java 7 for building and running. Any luck to upgrade it for duke 2.0?

**Comment by larsga** [comment]
Thank you: this is good work! We probably should abandon JDK 1.6 now (it was released in 2006) and instead switch to 1.7 and 1.8. I'll deal with that later on.

File: duke-core/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/.classpath
+/.settings/
```
Comment (drazzib) [inline duke-core/.gitignore]: I think we should update root .gitignore file with correct patterns and avoid maintaining .gitgnore for each submodule

File: duke-json/.classpath [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.m2e.MAVEN2_CLASSPATH_CONTAINER">
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>
```
Comment (drazzib) [inline duke-json/.classpath]: You should not commit .classpath file :)

File: pom.xml [hunk 3]
```diff
@@ -112,6 +113,11 @@ TO PUBLISH SNAPSHOTS
                 <artifactId>duke-json</artifactId>
                 <version>1.3-SNAPSHOT</version>
             </dependency>
+            <dependency>
+                <groupId>no.priv.garshol.duke</groupId>
+                <artifactId>duke-es</artifactId>
```
Comment (drazzib) [inline pom.xml]: You should also update assembly configuration duke-dist/src/main/assembly/dep.xml to include your new module into Duke distribution

**Comment by drazzib** [comment]
Thanks @fabriziofortino for this nice Elasticsearch module ! I just posted some small comments on individual commit.

**Comment by fabriziofortino** [comment]
@drazzib Thanks for reviewing it! I will update the PR as soon as possible

**Pull request merged**